metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -270,7 +353,19 @@ public TableDto deleteAndReturn(final QualifiedName name, final boolean isMView)
         // Try to delete the table even if get above fails
         try {
             connectorTableServiceProxy.delete(name);
```

Comment:
For drop, do we want to have `parentChildRelMetadataService.drop` run before `connectorTableServiceProxy.delete(name)`?

There are two pros for running `parentChildRelMetadataService.drop` first:
1. Validation that prevents dropping the children is actually performed in `parentChildRelMetadataService.drop`, and normally it makes sense to perform validation before taking certain action.
2. Dropping and then recover a table if there is any failures seems more destructive than dropping the parent child relation and then recovering the parent child relation.

Cons:
1. Race condition, between `parentChildRelMetadataService.drop` and `connectorTableServiceProxy.delete(name)`, a new clone table is created

Or we can always have a validation (read the children list and make sure there are no children) before `parentChildRelMetadataService.drop`, and then still keep  validation in `parentChildRelMetadataService.drop` as a safety guard.